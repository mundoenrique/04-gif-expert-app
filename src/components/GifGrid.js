// import { useEffect, useState } from 'react';

import GifGridItem from './GifGridItem';
import useFetchGifts from '../hooks/useFetchGifts';

export default function GifGrid({ category }) {
	const { data: images, loading } = useFetchGifts(category);

	return (
		<>
			<h3>{category}</h3>
			{loading && <p>Cargando...</p>}
			<div className="card-grid">
				{
					images.map(img => (
						<GifGridItem
							key={img.id}
							{...img}
						/>
					))
				}
			</div>
		</>
	)
}
