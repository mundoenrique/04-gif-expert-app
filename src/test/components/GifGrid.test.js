import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import useFetchGifs from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas al componene <GifGrid />', () => {
	const category = 'Batman'

	test('Debe mostrar el componente correctamente', () => {
		useFetchGifs.mockReturnValue({
			data: [],
			loading: true
		})
		const wrapper = shallow(<GifGrid category={category} />);

		expect(wrapper).toMatchSnapshot();
	});

	test('Debe mostrar los items cuando se cargan imágenes useFetchGifs', () => {
		const gifs = [{
			id: 'Abecd',
			url: 'https://localhost/image/img.jpg',
			title: 'Un título'
		}];

		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false
		});

		const wrapper = shallow(<GifGrid category={category} />);

		expect(wrapper).toMatchSnapshot();
	});
});
