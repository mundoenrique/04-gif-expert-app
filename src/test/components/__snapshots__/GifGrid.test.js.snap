// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componene <GifGrid /> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Batman
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas al componene <GifGrid /> Debe mostrar los items cuando se cargan imágenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Batman
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="Abecd"
      key="Abecd"
      title="Un título"
      url="https://localhost/image/img.jpg"
    />
  </div>
</Fragment>
`;
